auth/src/ios/credential_ios.mm

```diff
@@ -166,9 +166,16 @@ @implementation PhoneListenerDataObjC
   FIREBASE_ASSERT(future_api != nullptr);
 
   const auto handle = future_api->SafeAlloc<Credential>(kCredentialFn_GameCenterGetCredential);
+  /**
+   Linking GameKit.framework without using it on macOS results in App Store rejection.
+   Thus we don't link GameKit.framework to our SDK directly. `optionalLocalPlayer` is used for
+   checking whether the APP that consuming our SDK has linked GameKit.framework. If not, will
+   complete with kAuthErrorInvalidCredential error.
+   **/
+  GKLocalPlayer *_Nullable optionalLocalPlayer = [[NSClassFromString(@"GKLocalPlayer") alloc] init];
 
   // Early-out if GameKit is not linked
-  if ([GKLocalPlayer class] == nullptr) {
+  if (optionalLocalPlayer == nullptr) {
```

Comment:
nit: though it might not matter too much, but let's not mix with `nil` and `nullptr` here.

Use the following instead:
```
if (!optionalLocalPlayer) {
```

Reply:
Done